Model/src/main/java/org/apidb/apicommon/model/report/ai/expression/DailyCostMonitor.java

```diff
@@ -0,0 +1,141 @@
+package org.apidb.apicommon.model.report.ai.expression;
+
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.Writer;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.util.Optional;
+
+import org.gusdb.fgputil.FormatUtil;
+import org.gusdb.fgputil.IoUtil;
+import org.gusdb.fgputil.cache.disk.DirectoryLock;
+import org.gusdb.wdk.model.WdkModel;
+import org.gusdb.wdk.model.WdkModelException;
+import org.gusdb.wdk.model.WdkRuntimeException;
+import org.json.JSONObject;
+
+import com.openai.models.CompletionUsage;
+
+public class DailyCostMonitor {
+
+  private static final String DAILY_COST_ACCUMULATION_FILE = "daily_cost_accumulation.txt";
+
+  // model prop keys
+  private static final String MAX_DAILY_DOLLAR_COST_PROP_NAME = "MAX_DAILY_AI_EXPRESSION_DOLLAR_COST";
+  private static final String PLAN_COST_PER_1M_INPUT_TOKENS_PROP_NAME = "COST_PER_1M_AI_INPUT_TOKENs";
+  private static final String PLAN_COST_PER_1M_OUTPUT_TOKENS_PROP_NAME = "COST_PER_1M_AI_OUTPUT_TOKENS";
+
+  // lock characteristics
+  private static final long DEFAULT_TIMEOUT_MILLIS = 1000;
+  private static final long DEFAULT_POLL_FREQUENCY_MILLIS = 50;
+
+  // json props
+  private static final String JSON_DATE_PROP = "currentDate";
+  private static final String JSON_COST_PROP = "accumulatedCost";
+
+  // completion usage object representing 0 cost
+  private static final CompletionUsage EMPTY_COST = CompletionUsage.builder()
+      .promptTokens(0)
+      .completionTokens(0)
+      .build();
+
+  public static class CostExceededException extends RuntimeException { }
+
+  private final Path _costMonitoringDir;
+  private final Path _costMonitoringFile;
+
+  private double _maxDailyDollarCost;
+  private double _costPerInputToken;
+  private double _costPerOutputToken;
+
+  public DailyCostMonitor(WdkModel wdkModel) throws WdkModelException {
+    _costMonitoringDir = AiExpressionCache.getAiExpressionCacheParentDir(wdkModel);
```

Comment:
Did you mean to use the entire expression cache directory, an not create a subdirectory within it?